src/clj_nakadi_java/core.clj

```diff
@@ -47,19 +48,35 @@
     (typeLiteral [_] TypeLiterals/OF_STRING)))
 
 
-(defn ->subscription-stream-config [subscription-id-or-stream-config]
-  (if (string? subscription-id-or-stream-config)
-    (-> (StreamConfiguration.)
-        (.subscriptionId subscription-id-or-stream-config))
-    (do
-      (assert (.subscriptionId subscription-id-or-stream-config)
-              "Stream configuration must include subscriptionId.")
-      subscription-id-or-stream-config)))
+(defn method-exists? [method]
+  (first (filter #(= (str (:name %)) method) (:members (r/reflect StreamConfiguration)))))
 
 
-(defn consume-subscription [client subscription-id-or-stream-config callback]
-  (let [stream-config    (->subscription-stream-config subscription-id-or-stream-config)
-        stream-processor (-> client
+(defn ->camel-case [kebab]
+  (let [a (clojure.string/split (name kebab) #"\-")]
+    (str (first a) (apply str (map clojure.string/capitalize (rest a))))))
+ 
+
+(defn str-invoke [instance method-str & args]
+  (clojure.lang.Reflector/invokeInstanceMethod
+   instance
+   method-str
+   (to-array args)))
+
+
+(defn set-param [streamConfigInstance k v]
+  (let [method (->camel-case k)]
+    (if (method-exists? method)
+      (str-invoke streamConfigInstance method v)
```

Comment:
does str-invoke always return streamConfigInstance? if it's not, then the following calls will break

Reply:
I'd rather

```clj
(defn set-param [streamConfigInstance k v]
  (let [method (->camel-case k)]
    (when (method-exists? method)
      (str-invoke streamConfigInstance method v))
    streamConfigInstance))
```

Reply:
You are right, this way is safer.